x/authenticator/authenticator/message_filter_test.go

```diff
@@ -0,0 +1,135 @@
+			true,
+		},
+
+		{"bank send. bad sender",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{"from_address":"someoneElse","to_address":"to"}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("foo", 100))},
+			false,
+		},
+
+		{"bank send. any",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("foo", 100))},
+			true,
+		},
+
+		{"bank send. bad amount",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{"from_address":"from","to_address":"to", "amount": [{"denom": "foo", "amount": "50"}]}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("foo", 100))},
+			false,
+		},
+
+		{"bank send. amount as number",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{"from_address":"from","to_address":"to", "amount": [{"denom": "foo", "amount": 100}]}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("foo", 100))},
+			false, // This fails because of floats. Should be prevented by validation
+		},
+
+		{"bank send. amount as mix string number",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{"from_address":"from","to_address":"to", "amount": [{"denom": "foo", "amount": "100"}]}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("foo", 100))},
+			true,
+		},
+
+		{"bank send. amount as mix string number but bad",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{"from_address":"from","to_address":"to", "amount": [{"denom": "foo", "amount": "50"}]}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("foo", 100))},
+			false,
+		},
+
+		{"bank send. just denom",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{"from_address":"from","to_address":"to", "amount": [{"denom": "foo"}]}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("foo", 100))},
+			true,
+		},
+
+		{"bank send. just denom",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{"from_address":"from","to_address":"to", "amount": [{"denom": "foo"}]}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("foo", 1))},
+			true,
+		},
+
+		{"bank send. bad denom",
+			`{"type":"/cosmos.bank.v1beta1.MsgSend","value":{"from_address":"from","to_address":"to", "amount": [{"denom": "foo"}]}}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("bar", 100))},
+			false,
+		},
+
+		{"bank send. any match",
+			`{}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("bar", 100))},
+			true,
+		},
+
+		{"bank send. using map as generic",
+			`{{"type":"/cosmos.bank.v1beta1.MsgSend","value":{}}`,
```

Comment:
There's an extra `{` here, that should removed

Reply:
hmm, interesting that that test passes 🤔

Reply:
ahh, it didn't pass. Great! fixed now